projector-awt-common/src/main/kotlin/org/jetbrains/projector/awt/PToolkitUtils.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2019-2021, JetBrains s.r.o. and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation. JetBrains designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact JetBrains, Na Hrebenech II 1718/10, Prague, 14000, Czech Republic
+ * if you need additional information or have any questions.
+ */
+@file:Suppress("JAVA_MODULE_DOES_NOT_EXPORT_PACKAGE")
+
+package org.jetbrains.projector.awt
+
+import sun.awt.AWTAutoShutdown
+
+object PToolkitUtils {
+  private val unregisterPeerMethod = AWTAutoShutdown::class.java.getDeclaredMethod("unregisterPeer", Any::class.java,
+                                                                                   Any::class.java).apply {
+    isAccessible = true
+  }
+
+
+  fun targetDisposedPeer(target: Any, peer: Any) {
+    unregisterPeerMethod.invoke(AWTAutoShutdown.getInstance(), target, peer)
```

Comment:
There exists a `SunToolkit#targetDisposedPeer` which is `protected static` and thus accessible from `PToolkit`, same for `targetCreatedPeer`. Perhaps it would make sense to use those?

Reply:
`PToolkit` and `PComponentPeerBase` (where `targetDisposedPeer` is used) are in different modules (module of `PToolkit` dpends on module of `PComponentPeerBase`), so no way to call it without reflection. `targetCreatedPeer` on the other hand can be used